check_diff/src/lib.rs

```diff
@@ -80,8 +106,59 @@ impl RustfmtRunner {
         let binary_version = std::str::from_utf8(&command.stdout)?.trim();
         return Ok(binary_version.to_string());
     }
+
+    //  Run rusfmt with the --check flag to see if a diff is produced. Runs on the code specified
+    //
+    // Parameters:
+    // binary_path: Path to a rustfmt binary
+    // code: Code to run the binary on
+    // config: Any additional configuration options to pass to rustfmt
+    //
+    fn format_code<'a>(
+        &self,
+        code: &'a str,
+        config: &Option<Vec<String>>,
+    ) -> Result<String, CheckDiffError> {
+        let config = create_config_arg(config);
+        let mut command = Command::new(&self.binary_path)
+            .env("LD_LIBRARY_PATH", &self.ld_library_path)
+            .args([
+                "--unstable-features",
+                "--skip-children",
+                "--emit=stdout",
+                config.as_str(),
+            ])
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        command.stdin.as_mut().unwrap().write_all(code.as_bytes())?;
+        let output = command.wait_with_output()?;
+        Ok(std::str::from_utf8(&output.stdout)?.to_string())
+    }
 }
 
+fn create_config_arg(config: &Option<Vec<String>>) -> String {
+    let config_arg: String = match config {
+        Some(configs) => {
+            let mut result = String::new();
+            result.push(',');
+            for arg in configs.iter() {
+                result.push_str(arg.as_str());
+                result.push(',');
+            }
+            result.pop();
+            result
+        }
+        None => String::new(),
+    };
+    let config = format!(
+        "error_on_line_overflow=false,error_on_unformatted=false{}",
```

Comment:
I believe we need to start this with `--config=`
```suggestion
        "--config=error_on_line_overflow=false,error_on_unformatted=false{}",
```